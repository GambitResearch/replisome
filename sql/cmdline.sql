\set VERBOSITY terse

-- predictability
SET synchronous_commit = on;

SELECT 'init' FROM pg_create_logical_replication_slot('regression_slot', 'wal2json');

-- Unknown option
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL,
	'nosuchopt', '42');

-- Regexp error
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL,
	'include-table', '~(');

-- By default don't write in chunks
CREATE TABLE x ();
DROP TABLE x;
SELECT data FROM pg_logical_slot_peek_changes('regression_slot', NULL, NULL, 'include-xids', 'f', 'include-empty-xacts', '1');
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-xids', 'f', 'write-in-chunks', 't', 'include-empty-xacts', '1');

-- By default don't write xids
CREATE TABLE gimmexid (id integer PRIMARY KEY);
INSERT INTO gimmexid values (1);
DROP TABLE gimmexid;
SELECT max(((data::json) -> 'xid')::text::int) < txid_current() FROM pg_logical_slot_peek_changes('regression_slot', NULL, NULL, 'include-xids', '1', 'include-empty-xacts', '1');
SELECT max(((data::json) -> 'xid')::text::int) + 10 > txid_current() FROM pg_logical_slot_peek_changes('regression_slot', NULL, NULL, 'include-xids', '1', 'include-empty-xacts', '1');
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-empty-xacts', '1') WHERE ((data::json) -> 'xid') IS NOT NULL;

-- By default don't include empty transactions
CREATE TABLE emptyxact (id integer PRIMARY KEY);
INSERT INTO emptyxact values (1);
DROP TABLE emptyxact;
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL);

SELECT 'stop' FROM pg_drop_replication_slot('regression_slot');
