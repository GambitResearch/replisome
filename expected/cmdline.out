\set VERBOSITY terse
\pset format unaligned
-- predictability
SET synchronous_commit = on;
SELECT 'init' FROM pg_create_logical_replication_slot('regression_slot', 'replisome');
?column?
init
(1 row)
-- Unknown option
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL,
	'nosuchopt', '42');
ERROR:  option "nosuchopt" = "42" is unknown
-- Regexp error
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL,
	'include-table', '~(');
ERROR:  invalid regular expression "(": parentheses () not balanced
-- By default don't write in chunks
CREATE TABLE x ();
DROP TABLE x;
SELECT data FROM pg_logical_slot_peek_changes('regression_slot', NULL, NULL, 'include-xids', 'f', 'include-empty-xacts', '1');
data
{"change":[]}
{"change":[]}
(2 rows)
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-xids', 'f', 'write-in-chunks', 't', 'include-empty-xacts', '1');
data
{"change":[
]}
{"change":[
]}
(4 rows)
-- By default don't write xids
CREATE TABLE gimmexid (id integer PRIMARY KEY);
INSERT INTO gimmexid values (1);
DROP TABLE gimmexid;
SELECT max(((data::json) -> 'xid')::text::int) < txid_current() FROM pg_logical_slot_peek_changes('regression_slot', NULL, NULL, 'include-xids', '1', 'include-empty-xacts', '1');
?column?
t
(1 row)
SELECT max(((data::json) -> 'xid')::text::int) + 10 > txid_current() FROM pg_logical_slot_peek_changes('regression_slot', NULL, NULL, 'include-xids', '1', 'include-empty-xacts', '1');
?column?
t
(1 row)
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-empty-xacts', '1') WHERE ((data::json) -> 'xid') IS NOT NULL;
data
(0 rows)
-- By default don't include empty transactions
CREATE TABLE emptyxact (id integer PRIMARY KEY);
INSERT INTO emptyxact values (1);
DROP TABLE emptyxact;
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL);
data
{"change":[{"op":"I","schema":"public","table":"emptyxact","colnames":["id"],"coltypes":["int4"],"values":[1]}]}
(1 row)
SELECT 'stop' FROM pg_drop_replication_slot('regression_slot');
?column?
stop
(1 row)
